### PR TITLE
Update code

### PR DESCRIPTION
**Pull request by Master-Vova**
Adding base functions, login, calculations and writing 4 tests

### CODE REVIEW

File: IntegerDivision/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.13.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>RELEASE</version>
```
Comment (NikitaHarkov) [inline IntegerDivision/pom.xml]: Always use a stable version - preferably the latest one.
You can find it those in maven central repository e.g. 
https://mvnrepository.com/artifact/junit/junit

File: IntegerDivision/pom.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.example</groupId>
+    <artifactId>IntegerDivision</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
```
Comment (NikitaHarkov) [inline IntegerDivision/pom.xml]: My advice is to use the latest Java LTS. If you don't know what it is - read in google

File: IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package org.example;
+
+public class DivisionProcessUtil {
+
+    public static String process(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java]: It cannot be a Util class (and it’s not even a correctly defined one) because you’re dealing with specific business logic. Utility classes should only contain small, generic helper methods, not an entire processing mechanism. Instead of making it a Util, you should implement it as a Processor class, for example:
`DivisionProcessor processor = new DivisionProcessor();`

File: IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            visualResult = visualResult.concat(writingString(changingDividend, changingDivisor, digitsChecked));
+            digitsChecked++;
+        }
+        visualResult = visualResult.concat(" ".repeat(numberLength(dividend)
+                - numberLength(reminder) + 1) + reminder);
+
+        return visualResult;
+    }
+
+    protected static String firstWriteString(int initialDividend, int initialDivisor,
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java]: Package access should be defined by need. If you don't call it outside of the class - it is `private` then
https://www.baeldung.com/java-access-modifiers

File: IntegerDivision/src/test/java/org/example/DivisionProcessUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
+package org.example;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class DivisionProcessUtilTest {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/test/java/org/example/DivisionProcessUtilTest.java]: Don’t forget about negative test cases. For example, what happens when you pass 0 as the divisor? Your logic should also handle this scenario correctly.
Also, make sure to cover edge cases. How does your code behave when either `dividend` or `divisor` is set to `Integer.MAX_VALUE`? When dividend is less then divisor? Or even negative?

File: IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+public class DivisionProcessUtil {
+
+    public static String process(int dividend, int divisor) {
+
+        int digitsChecked = 0;
+        int reminder = 0;
+        String visualResult = "";
+
+        while (digitsChecked < numberLength(dividend)) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessUtil.java]: Currently, your code looks messy because you’ve put everything in one place. For example, you’re mixing division logic with string preparation. If I asked you to extend or add a feature to the string-building part, would it be easy to do?

Try to stick with OOP principles:https://www.baeldung.com/java-oop
In your case, the result of the calculation could be stored and processed separately, while a dedicated class should handle formatting and preparing the string output.

File: IntegerDivision/src/main/java/org/example/LongDivisionSimulator.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package org.example;
+
+import java.util.Scanner;
+
+public class LongDivisionSimulator {
+    public static void main(String[] args) {
+
+        Scanner write = new Scanner(System.in);
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/LongDivisionSimulator.java:8]: To ease your life, you can currently go without Scanner, but manually provide values to the methods.

File: IntegerDivision/src/main/java/org/example/CalculationMethodsHelper.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package org.example;
+
+class CalculationMethodsHelper {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/CalculationMethodsHelper.java]: 1. All utility classes should reside in the `utils `package and be declared `final `with a private constructor to prevent instantiation.
2.  Method names should be more descriptive. Use the prefix `get `for methods that return a value. The class name should be concise so that method calls are intuitive, e.g. `DigitsUtils.getNumberLength()`.
3. Methods should have clear visibility - use `public `for externally used utilities and `private `for internal helpers. Avoid `protected `in utility classes.
4. All inputs should have validations. Since you're using primitive types, so at least check that those are not negative or zero, if you're not expecting those

File: IntegerDivision/src/main/java/org/example/CalculationMethodsHelper.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        }
+        return result;
+    }
+
+    protected static int numberLength(int dividend) {
+        String sample = Integer.toString(dividend);
+        return sample.length();
+    }
+
+    protected static int checkIfNumberLengthHasDifference(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/CalculationMethodsHelper.java]: Why don't you use boolean in this case? 
You can simply `return numberLength(dividend) > numberLength(divisor)`
And name method as `lengthIsDifferent` / `differentLength`

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            throw new IllegalArgumentException("Calculation with negative numbers is still developing... " +
+                    "Sorry for the inconveniences");
+        } else if (divisor == 0) {
+            throw new IllegalArgumentException("Dividing by zero isn't allowed");
+        } else if (dividend < divisor) {
+            throw new IllegalArgumentException("Dividend is less than divisor...");
+        }
+    }
+
+    private static int currentDigit(int dividend, int i) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Why you have new methods instead of directly using utility class?

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+public class DivisionProcessor {
+    private DivisionProcessor() {
+    }
+
+    public static String process(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Method names like process are too generic. Rename it to something more descriptive, e.g. `performDivision()` or `buildLongDivisionVisualization()` depending on the actual purpose.

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+                digitsChecked++;
+                continue;
+            } else if (visualResult.equals("")) {
+                visualResult = visualResult.concat(StringConstuctor.firstWriteString(dividend,
+                        divisor, changingDividend, changingDivisor));
+                digitsChecked++;
+                continue;
+            }
+
+            visualResult = visualResult.concat(StringConstuctor.writingString(changingDividend,
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Using String.concat() repeatedly in a loop is inefficient. Replace with a StringBuilder for better performance.

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+            int changingDividend = (reminder * 10) + currentDigit(dividend, digitsChecked);
+            int changingDivisor = (changingDividend / divisor) * divisor;
+            reminder = changingDividend % divisor;
+
+            if (changingDividend < divisor) {
+                digitsChecked++;
+                continue;
+            } else if (visualResult.equals("")) {
+                visualResult = visualResult.concat(StringConstuctor.firstWriteString(dividend,
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Instead of returning a formatted string directly, consider returning an object (e.g. DivisionResult) that contains all relevant values and the result of the division.
This object can then be passed to StringConstructor (or a similar formatter) for string formatting.
This approach separates computation from presentation and makes the code more reusable and testable.

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+                    changingDivisor, digitsChecked));
+            digitsChecked++;
+        }
+        visualResult = visualResult.concat(" ".repeat(numberLength(dividend)
+                - numberLength(reminder) + 1) + reminder);
+
+        return visualResult;
+    }
+
+    private static void inputCheck(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Usually, these methods are named as `validate`

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 15]
```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+public class DivisionProcessor {
+    private DivisionProcessor() {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: This class is not a utility class - it contains business logic rather than generic, reusable helper methods.
Utility classes are meant for stateless, shared functionality across the application.
Please refactor this to be a regular (non-static) class that can be instantiated or managed as part of the business layer.

File: IntegerDivision/src/main/java/org/example/DivisionProcessor.java [hunk 16]
```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+public class DivisionProcessor {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionProcessor.java]: Probably better naming would be IntegerDivision in this case

File: IntegerDivision/src/main/java/org/example/StringConstuctor.java [hunk 17]
```diff
@@ -0,0 +1,44 @@
+package org.example;
+
+class StringConstuctor {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/StringConstuctor.java]: Same comment regards util class as for DivisionProcessor.
And better naming for this class will be something like DivisionFormatter. StringConstuctor sounds bad

File: IntegerDivision/src/main/java/org/example/StringConstuctor.java [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+        int dif = checkIfNumberLengthHasDifference(dividend, divisor);
+
+        String line1 = String.format("%s_%s\n", " ".repeat(x - 1), dividend);
+        String line2 = String.format("%s%s\n", " ".repeat(x + dif), divisor);
+        String line3 = String.format("%s%s\n", " ".repeat(x + dif), "-".repeat(numberLength(divisor)));
+
+        return line1 + line2 + line3;
+    }
+
+    private static int numberLength(int dividend) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/StringConstuctor.java]: Same as for DivisionProcessor -  Why do you have new methods instead of directly using the utility class?

File: IntegerDivision/src/main/java/org/example/StringConstuctor.java [hunk 19]
```diff
@@ -0,0 +1,44 @@
+package org.example;
+
+class StringConstuctor {
+    private StringConstuctor() {
+    }
+
+    protected static String firstWriteString(int initialDividend, int initialDivisor,
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/StringConstuctor.java]: You are manually concatenating strings using + and String.format() multiple times. Consider using a StringBuilder for better performance and readability.

**Review by NikitaHarkov — CHANGES_REQUESTED** [review]

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 20]
```diff
@@ -0,0 +1,38 @@
+package org.example;
+
+final class DigitsUtil {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: Why is this class using the default (package-private) access modifier?
If there’s no specific reason to restrict visibility, it’s generally better to make it public for clarity and consistency.
The same goes for methods
Consider this change also in other classes

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+    private DigitsUtil() {
+    }
+
+    static int getDigitFromNumber(int dividend, int i) {
+        validate(dividend);
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: You can modify the validate method to accept multiple parameters, e.g.
` private static void validate(int... number) {}`
Then iterate through all numbers inside the method.
This way, you can call it like `validate(dividend, i)` and avoid calling the method twice.

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+     static  boolean lengthItDifferent(int dividend, int divisor) {
+        validate(dividend);
+        validate(divisor);
+
+        return getNumberLength(dividend) > getNumberLength(divisor);
+    }
+
+    private static void validate(int number) {
+        if (number < 0) {
+            throw new IllegalArgumentException("Utility class got wrong input");
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: The error message `Utility class got wrong input` is too generic.
Please make it more specific - it should clearly describe why the operation failed and include contextual details (e.g. the invalid number or parameter and why it’s invalid).

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 23]
```diff
@@ -0,0 +1,40 @@
+package org.example;
+
+public class IntegerDivision {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java:3]: You're moving in the right direction now, but still not ideal. Currently, your class holds 2 responsibilities. One is performing division, the second is writing a String. 
In OOP, it’s best to follow the SOLID principles (see [link](https://contabo.com/blog/what-are-solid-principles/)), where the first “S” stands for Single Responsibility.
Each class should have only one clear purpose, so consider separating the division logic from the output handling.

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+            }
+            formatter.writeString(changingDividend, changingDivisor, digitsChecked);
+            digitsChecked++;
+        }
+        formatter.writeLastLine(reminder);
+
+        return formatter.result.toString();
+    }
+
+    private static void validate(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java]: When throwing errors, if the input is not sensitive information, include the actual value that caused the failure.
This makes debugging much easier by clearly showing which value triggered the error.

**Review by NikitaHarkov — CHANGES_REQUESTED** [review]

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java (hunk 23 above)
Comment (Master-Vova) [reply IntegerDivision/src/main/java/org/example/IntegerDivision.java:3]: Tried to split responsibilities by transferring calculation part to new class.

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java (hunk 21 above)
Comment (Master-Vova) [reply IntegerDivision/src/main/java/org/example/DigitsUtil.java]: Fixed.

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+
+     public static boolean lengthItDifferent(int dividend, int divisor) {
+        validate(dividend, divisor);
+        return getNumberLength(dividend) > getNumberLength(divisor);
+    }
+
+    private static void validate(int... numbers) {
+        for (int singleNumber : numbers) {
+            if (singleNumber < 0) {
+                throw new IllegalArgumentException("Utility class got negative input");
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: `"Number is negative " + singleNumber` will be better

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+     public static boolean lengthItDifferent(int dividend, int divisor) {
+        validate(dividend, divisor);
+        return getNumberLength(dividend) > getNumberLength(divisor);
+    }
+
+    private static void validate(int... numbers) {
+        for (int singleNumber : numbers) {
+            if (singleNumber < 0) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: Naming such variables as `number` is fine. Even as `n`. No need for such clarity as `single `prefix

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 27]
```diff
@@ -0,0 +1,37 @@
...
+        if (result >= 10) {
+            result = result % 10;
+        }
+        return result;
+    }
+
+    public static int getNumberLength(int number) {
+        validate(number);
+
+        String sample = Integer.toString(number);
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: You can avoid declaring unnecessary variable and just return the result as `return Integer.toString(number).length()`

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+            digitsChecked++;
+        }
+        formatter.writeLastLine(calculation.reminder);
+
+        return formatter.result.toString();
+    }
+
+    private static void validate(int dividend, int divisor) {
+        if (dividend < 0) {
+            throw new IllegalArgumentException(dividend + " value isn't allowed...");
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java]: Always provide info on why exactly it isn't allowed. In this case `... value less than zero isn't allowed`

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 29]
```diff
@@ -0,0 +1,41 @@
+package org.example;
+
+public class IntegerDivision {
+
+    public String buildLongDivisionVisualization(int dividend, int divisor) {
+
+        validate(dividend, divisor);
+
+        DivisionFormatter formatter = new DivisionFormatter(dividend, divisor);
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java]: You didn’t fully address my previous comment. In the current state, your class still has unnecessary dependencies and the main issue is that they are declared inside the method instead of being properly initialized in the class, which is incorrect.

- Calculation class
- DisivisonFormatter class

But my main point, which I mentioned earlier, is that you should separate the logic.
First, perform the calculation (including all validation) and obtain the result. Then, pass this result to a separate method in the `DivisionFormatter `class, which should be responsible only for formatting the output.

You shouldn’t combine all logic in one place - that approach quickly leads to messy and unmaintainable code.

Here’s how your `LongDivisionSimulator `class should ideally look:
```
IntegerDevision integerDevision = new IntegerDevision()
DevisionResult result = integerDevision.calculate(dividend, divisor);

DivisionFormatter formatter = new DivisionFormatter();
String resultString = formatter.formatDivisionResult(result);

System.out.println(resultString);
```

**Review by NikitaHarkov — CHANGES_REQUESTED** [review]

File: IntegerDivision/src/main/java/org/example/DigitsUtil.java [hunk 30]
```diff
@@ -0,0 +1,36 @@
+package org.example;
+
+final public class DigitsUtil {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DigitsUtil.java]: In Java, the usual modifier order is `public fina`l rather than `final public`. Please swap the modifiers for consistency.

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 31]
```diff
@@ -0,0 +1,45 @@
+package org.example;
+
+public class IntegerDivision {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java:3]: Your `DivisionFormatter `class uses the `default `visibility modifier, while `IntegerDivision `is declared as `public`. Please align them unless there's a specific reason for the difference.

File: IntegerDivision/src/main/java/org/example/DivisionFormatter.java [hunk 32]
```diff
@@ -0,0 +1,65 @@
+package org.example;
+
+class DivisionFormatter{
+    private int initialDividend;
+    private int initialDivisor;
+    private StringBuilder resultString = new StringBuilder();
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionFormatter.java]: These three fields (initialDividend, initialDivisor, and resultString) make the class stateful, although the logic works like a pure formatting function. Since they are only used within a single call to formatResult(), it’s better to convert them into local variables and pass them as needed. This will make the class thread-safe, easier to reason about, and more aligned with functional-style design.

File: IntegerDivision/src/main/java/org/example/DivisionFormatter.java [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+
+    private void writeStringForFirstTime(int changingDividend, int changingDivisor) {
+
+        int quotient = initialDividend / initialDivisor;
+        int dif = DigitsUtil.lengthItDifferent(changingDividend, changingDivisor) ? 1 : 0;
+        String spaceAfter = " ".repeat(DigitsUtil.getNumberLength(initialDividend) -
+                dif - DigitsUtil.getNumberLength(changingDivisor));
+        String spaceBefore = " ".repeat(dif + 1);
+
+        resultString
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/DivisionFormatter.java]: Small improvement suggestion: avoid `"_" + initialDividend + "│" + initialDivisor `inside` append()`. When you concatenate strings, the compiler creates intermediate String instances. StringBuilder is meant to prevent that, so it's cleaner and more efficient to chain several append() calls instead.

File: IntegerDivision/src/main/java/org/example/IntegerDivision.java [hunk 34]
```diff
@@ -0,0 +1,45 @@
...
+
+            if (changingDividend < initialDivisor){
+                arr[0][digitsChecked] = 1;
+            }
+            digitsChecked++;
+        }
+        return new DivisionResult(initialDividend, initialDivisor, arr, reminder);
+    }
+
+    private static void validate(int dividend, int divisor) {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/main/java/org/example/IntegerDivision.java]: The` validate()` method is static, but the rest of the logic is instance-based. Static methods are harder to override, mock, or extend, so they limit flexibility if validation rules change in the future. Since validation is part of the calculation flow, it fits better as a private instance method.

File: IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DivisionProcessorTest {
+    @Test
+    public void testNumber1823000002DevidedBy13() {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java]: Including the concrete numbers in the test method name makes it long and harder to read. It’s typically better to use a descriptive name (e.g., `testNumberDivision`), while keeping the actual test values inside the method body.

File: IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java [hunk 36]
```diff
@@ -0,0 +1,64 @@
+package org.example;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DivisionProcessorTest {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java]: Here, you're currently testing only `IntegerDivision `class, so test class should be named accordingly

File: IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java [hunk 37]
```diff
@@ -0,0 +1,64 @@
...
+        String expected = "Dividend is less than divisor...";
+        IntegerDivision IntegerDivision = new IntegerDivision();
+
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () ->
+                IntegerDivision.calculate(14823, 20000));
+        assertEquals(expected, exception.getMessage());
+    }
+
+    @Test
+    public void testForArrays() {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/test/java/org/example/DivisionProcessorTest.java]: This test appears to duplicate the first positive-case test. Since the positive scenario is already verified, repeating it doesn't improve coverage. Consider replacing it with an edge case or removing it.

File: IntegerDivision/src/test/java/org/example/FormatterTests.java [hunk 38]
```diff
@@ -0,0 +1,143 @@
+package org.example;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class FormatterTests {
```
Comment (NikitaHarkov) [inline IntegerDivision/src/test/java/org/example/FormatterTests.java:8]: Please revise the test method names to be more general (no specific values in the method name). Since one positive test case is sufficient, additional tests should focus on negative or edge conditions. At the moment, negative scenarios are missing - what about them? 🙂

**Review by NikitaHarkov — CHANGES_REQUESTED** [review]

**Comment by NikitaHarkov** [comment]
LKFM! 
Good job!
